picard/util/thread.py

```diff
@@ -26,8 +26,8 @@
 
 
 import sys
-import traceback
 from threading import Event
+import traceback
```

Comment:
Prolly a leftover?

Reply:
The Event import? Not really. Using it to notify the threads that the main thread already ran a task before proceeding. It replaced the time.sleep() with arbitrary values. I'm replacing that with a __init__ parameter. It didn't look as bad when I was coding it.

```
class ProxyToMainEvent(QEvent):

    def __init__(self, func, *args, **kwargs):
        super().__init__(QEvent.User)
        self.func = func
        self.args = args
        self.kwargs = kwargs
        self.event = kwargs.pop('event') if 'event' in kwargs else None

    def run(self):
        self.func(*self.args, **self.kwargs)
        if self.event:
            self.event.set()
[...]

def to_main(func, *args, **kwargs):
    event = Event()
    kwargs['event'] = event
    QCoreApplication.postEvent(QCoreApplication.instance(),
                               ProxyToMainEvent(func, *args, **kwargs))
    return event

```

Reply:
I had previously suggested using a queue to wait on instead. Use of Event is equally architecturally sound, however I would personally like to see the main thread run all pending events before handing off execution back to a worker thread by doing event.set(). 

I also have a nagging worry (which may not be valid) that use of a self.event to store the event rather than passing it around as a parameter may lead to it being overwritten by another event - but as I say, this may simply be because I have not spent enough time understanding the scope of self.event to know that this is not possible.